File: 2Arrays and Strings/Code/Python/2-2Palindrome.py
def check_if_palindrome(s):
    left = 0
    right = len(s) - 1

    while left < right:
        if s[left] != s[right]:
            return False
        left += 1
        right -= 1
    
    return True

# Odd length strings
print(check_if_palindrome("racecar")) # True
print(check_if_palindrome("aslkd"))   # False

# Even length strings
print(check_if_palindrome("basdab"))   # False
print(check_if_palindrome("abassaba")) # True

# Time-complexity : O(n)
# Space complexity: O(1)

File: 2Arrays and Strings/Code/JavaScript/2-2Palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var checkIfPalindrome = function(s) {
    let left = 0;
    let right = s.length - 1;
    
    while (left < right) {
        if (s[left] != s[right]) {
            return false;
        }
        
        left++;
        right--;
    }
    
    return true;
}

// odd length strings
console.log(checkIfPalindrome("racecar")); // prints true
console.log(checkIfPalindrome("aslkd")); // prints false

// even length strings
console.log(checkIfPalindrome("basdab")); // prints false
console.log(checkIfPalindrome("abassaba")); // prints true

// Time-complexity : O(n)
// Space complexity: O(1)
